openedx/core/djangoapps/user_authn/views/login.py

```diff
@@ -110,6 +115,31 @@ def _get_user_by_email(request):
         AUDIT_LOG.warning(f"Login failed - Unknown user email {digest}")
 
 
+def _get_user_by_email_or_username(request):
+    """
+    Finds a user object in the database based on the given request, ignores all fields except for email and username.
+    """
+    if not (
+        'email' in request.POST or 'username' in request.POST
+    ) or 'password' not in request.POST:
+        raise AuthFailedError(_('There was an error receiving your login information. Please email us.'))
+
+    email = request.POST.get('email', None)
+    username = request.POST.get('username', None)
+
+    try:
+        return USER_MODEL.objects.get(
+            Q(username=username) | Q(email=email)
+        )
+    except USER_MODEL.DoesNotExist:
+        if email:
+            digest = hashlib.shake_128(email.encode('utf-8')).hexdigest(16)  # pylint: disable=too-many-function-args
+            AUDIT_LOG.warning(f"Login failed - Unknown user email {digest}")
+        else:
+            digest = hashlib.shake_128(username.encode('utf-8')).hexdigest(16)  # pylint: disable=too-many-function-args
+            AUDIT_LOG.warning(f"Login failed - Unknown username {digest}")
```

Comment:
This can be simplified.
```suggestion
        username_or_email = email or username
        digest = hashlib.shake_128(username_or_email.encode('utf-8')).hexdigest(16)  # pylint: disable=too-many-function-args
        AUDIT_LOG.warning(f"Login failed - Unknown user username/email {digest}")
```